src/main/java/org/swisspush/redisques/QueueStatsService.java

```diff
@@ -0,0 +1,219 @@
+                log.warn("queueStatisticsCollector.getQueueStatistics() failed. Fallback to empty result.", ev.cause());
+                req.queuesJsonArr = new JsonArray();
+                onDone.accept(null, req);
+                return;
+            }
+            JsonObject queStatsJsonObj = ev.result();
+            String status = queStatsJsonObj.getString(STATUS);
+            if (!OK.equals(status)) {
+                log.warn("queueStatisticsCollector.getQueueStatistics() responded '" + status + "'. Fallback to empty result.", ev.cause());
+                req.queuesJsonArr = new JsonArray();
+                onDone.accept(null, req);
+                return;
+            }
+            req.queuesJsonArr = queStatsJsonObj.getJsonArray(QUEUES);
+            onDone.accept(null, req);
+        });
+    }
+
+    private <CTX> void mergeRetryDetailsIntoCollectedData(GetQueueStatsRequest<CTX> req, BiConsumer<Throwable, GetQueueStatsRequest<CTX>> onDone) {
+        // Setup a lookup table as we need to find by name further below.
+        Map<String, JsonObject> detailsByName = new HashMap<>(req.queuesJsonArr.size());
+        for (var it = (Iterator<JsonObject>) (Object) req.queuesJsonArr.iterator(); it.hasNext(); ) {
+            JsonObject detailJson = it.next();
+            String name = detailJson.getString(MONITOR_QUEUE_NAME);
+            detailsByName.put(name, detailJson);
+        }
+        for (Queue queue : req.queues) {
+            JsonObject detail = detailsByName.get(queue.name);
+            if (detail == null) continue; // no details to enrich.
+            JsonObject dequeueStatsJson = detail.getJsonObject(STATISTIC_QUEUE_DEQUEUESTATISTIC);
+            if (dequeueStatsJson == null) continue; // no dequeue stats we could enrich
+            DequeueStatistic dequeueStats = dequeueStatsJson.mapTo(DequeueStatistic.class);
+            // Attach whatever details we got.
+            queue.lastDequeueAttemptEpochMs = dequeueStats.lastDequeueAttemptTimestamp;
+            queue.lastDequeueSuccessEpochMs = dequeueStats.lastDequeueSuccessTimestamp;
+            queue.nextDequeueDueTimestampEpochMs = dequeueStats.nextDequeueDueTimestamp;
+        }
+        onDone.accept(null, req);
+    }
+
+    private int compareLargestFirst(Queue aq, Queue bq) {
+        if (aq.size == null && bq.size == null) return 0;
+        if (aq.size == null) return -1;
+        if (bq.size == null) return +1;
+        long as = aq.size, bs = bq.size;
+        if (as > bs) return -1;
+        if (as < bs) return +1;
+        assert as == bs : as +", "+ bs;
+        return 0;
+    }
+
+
+    private static class GetQueueStatsRequest<CTX> {
+        private CTX mCtx;
+        private GetQueueStatsMentor<CTX> mentor;
+        private List<String> queueNames;
+        private JsonArray queuesJsonArr;
+        private List<Queue> queues;
+    }
+
+
+    public static class Queue {
+        private String name;
+        private Long size;
```

Comment:
This should be a primitive `long` for the following reasons:
- A queue always has a size `>=0` (either it has some elements in it or it doesn't)
- `getSize()` returns a primitive (could lead to NullPointerException)
- Making it a primitive would simplify `compareLargestFirst()`.